src/service/viewer-impl.js

```diff
@@ -860,12 +860,33 @@ export class Viewer {
     if (!this.hasCapability('fragment')) {
       return Promise.resolve('');
     }
-    return this.sendMessageUnreliable_('fragment', undefined, true).then(
+    return this.sendMessageUnreliable_('getFragment', undefined, true).then(
       hash => hash || ''
     );
   }
 
   /**
+   * Update the fragment of the viewer if embedded in a viewer,
+   * otherwise update the page url fragment
+   * @param {string} fragment
+   * @return {!Promise}
+   */
+  updateFragment(fragment) {
+    if (!this.isEmbedded_) {
+      if (this.win.history.replaceState) {
+        const newUrl = this.win.location.href.split('#')[0] + '#' + fragment;
+        this.win.history.replaceState({}, '', newUrl);
+      }
+      return Promise.resolve();
+    }
+    if (!this.hasCapability('fragment')) {
+      return Promise.resolve();
+    }
+    return /** @type {!Promise} */ (this.sendMessageUnreliable_(
```

Comment:
> Did you read #3135?
  
Yah, but that doesn't have anything related to the Google side.

> www.google.com/amp/some-publisher/some-page.html#.12345

Perfect, for some reason I thought the share links were just the iframe's window's href. 😳